cmds/wget/wget_test.go

```diff
@@ -103,42 +85,52 @@ func getFreePort(t *testing.T) int {
 
 // TestWget implements a table-driven test.
 func TestWget(t *testing.T) {
-	tmpDir, execPath := testutil.CompileInTempDir(t)
-	defer os.RemoveAll(tmpDir)
-
 	// Start a webserver on a free port.
-	port := getFreePort(t)
 	unusedPort := getFreePort(t)
-	h := handler{}
+
+	server, err := net.Listen("tcp", ":0")
+	if err != nil {
+		t.Fatalf("Cannot get free port for server: %v", err)
+	}
+	port := server.Addr().(*net.TCPAddr).Port
 	go func() {
-		t.Fatal(http.ListenAndServe(fmt.Sprintf(":%d", port), h))
+		t.Fatal(http.Serve(server, handler{}))
 	}()
 
-	time.Sleep(500 * time.Millisecond) // TODO: better synchronization
 	for i, tt := range tests {
-		// Arguments inherited from the environment.
-		execArgs := strings.Split(os.Getenv("EXECPATH"), " ")[1:]
+		t.Run(fmt.Sprintf("Test [%02d]", i), func(t *testing.T) {
+			execPath := os.Getenv("EXECPATH")
+			uri := fmt.Sprintf(tt.url, port, unusedPort)
+
+			var err error
+			var out string
 
-		args := append(append(execArgs, tt.flags...), fmt.Sprintf(tt.url, port, unusedPort))
-		out, err := exec.Command(execPath, args...).Output()
+			if len(execPath) > 0 {
+				// Arguments inherited from the environment.
+				execArgs := strings.Fields(execPath)
+				args := append(execArgs[1:], uri)
+				cmd := exec.Command(execArgs[0], args...)
```

Comment:
The logic here is similar to `CompileInTempDir`. Are you not using it because you want code coverage?

Reply:
I don't want to use the compiled wget. I either want to use exec path or the already compiled in `wget` function. Maybe I can factor this out into some test utility tough.

Reply:
Sounds like a plan!